src/main/org/codehaus/groovy/runtime/DefaultGroovyMethods.java

```diff
@@ -12026,6 +12026,9 @@ public static boolean equals(List left, List right) {
         if (left == right) {
             return true;
         }
+        if( left.getClass().getAnnotation(IgnoreDefaultEqualsAndToString.class)!=null && right.getClass().getAnnotation(IgnoreDefaultEqualsAndToString.class)!=null ) {
```

Comment:
-1 to this until we have a better idea on the impact on performance. Calling `getClass().getAnnotation(...)` is **very** expensive.

Reply:
Could you suggest a possible metrics/benchmark ?

Reply:
I have nothing particular in mind at this point, but it involves this method being called with different kind of objects, and comparing with the baseline without the change.